### PR TITLE
DDS-908 Add breadcrumb component

### PR DESCRIPTION
**Pull request by rinjInTokyo**
This PR adds breadcrumb and breadcrumb-item (`daikin-breadcrumb`, `daikin-breadcrumb-item`) component.

## Related Issue(s) and Links

- JIRA Ticket (parent): [DDS-908](https://daikinsvoil.atlassian.net/browse/DDS-908)
- Figma design: [Figma](https://www.figma.com/design/VyaaU8Ta9yzyf0PsURWSSf/branch/wHAhw4yC54Hjq6Whb8FL5T/DDS%3A-Design-Kit?node-id=23120-920&m=dev)

## API Design

[JIRA Wiki]Will add after

## How Has This Been Tested?

- breadcrumb
  - VRT (Visual Regression Test) for \[withTrailingSlash; withoutTrailingSlash\] trailingSlash style and \[withOmission; withoutOmission\] omission style.
  - If omission is true and the breadcrumb will be omitted when wrap container becomes smaller
- breadcrumb-item
  - VRT (Visual Regression Test) for \[max; min\] variants and \[Normal; Hover; Pressed; Disabled\]  state

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have added appropriate VRTs.
- [x] I have added appropriate unit tests.
- [x] I have confirmed that my changes do not generate new warnings and errors.

[DDS-908]: https://daikinsvoil.atlassian.net/browse/DDS-908?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by hiro-daikin — APPROVED** [review]
Looks good for now. 

One thing we may want to consider is adding an ellipsis at the end of each item. For example, `Home -> Item 1 -> Really long str....`

File: src/components/breadcrumb-item/daikin-breadcrumb-item.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      variant: this.variant,
+      disabled: this.disabled,
+    });
+    const slash = this.trailingSlash
+      ? html`<span class="text-daikinNeutral-800 font-daikinSerif">/</span>`
+      : html``;
+    const linkText =
+      this.variant === "normal"
+        ? html`<slot></slot>`
+        : html`<span>. . .</span>`;
```
Comment (yodas7) [inline src/components/breadcrumb-item/daikin-breadcrumb-item.ts:109]: Depending on our needs, we need to change the code depending on what we want the "text on the display", the "text to be selected (copied)", and the "text to be read by the screen reader" to be.
At least for screen readers, "…" or "..." should be better than ". . .", so `aria-label` should be specified.

```suggestion
        : html`<span aria-label="…">. . .</span>`;
```

File: src/components/breadcrumb-item/daikin-breadcrumb-item.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    return html`
+      <slot name="link"
+        ><a
+          href="${ifDefined(this.href)}"
+          class="${linkClassName}"
+          target="${this.target}"
+          >${linkText}</a
+        >${slash}</slot
+      >
```
Comment (yodas7) [inline src/components/breadcrumb-item/daikin-breadcrumb-item.ts:118]: nit.

```suggestion
      <slot name="link">
        <a
          href=${ifDefined(this.href)}
          class=${linkClassName}
          target=${this.target}
        >${linkText}</a>
        ${slash}
      </slot>
```

File: src/components/breadcrumb-item/daikin-breadcrumb-item.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+   * Specify whether the link should be disabled
+   */
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  /**
+   * Specify the link target
+   */
+  @property({ type: String, reflect: true })
+  target: "_blank" | "_self" | "_parent" | "_top" | "framename" = "_self";
```
Comment (yodas7) [inline src/components/breadcrumb-item/daikin-breadcrumb-item.ts:84]: `target` actually accepts any string, as "framename" is not just "framename" but is a frame's name.

```suggestion
  target: string | null = null;
```

File: src/components/breadcrumb/daikin-breadcrumb.ts [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    } else if (
+      (breadcrumbWidth > this.divOriginalWidth || !this._isEllipsis) &&
+      this.omissionMode
+    ) {
+      daikinBreadCrumbItems?.forEach((value: Element) => {
+        value.setAttribute("variant", "normal");
+        value.removeAttribute("hidden");
+      });
+      this.omissionMode = false;
+    }
```
Comment (yodas7) [inline src/components/breadcrumb/daikin-breadcrumb.ts:99]:
```suggestion
    const shouldOmit = this.overflow === "ellipsis" && this.offsetWidth < this._expandedContentWidth;
    if (this.omitted === shouldOmit) {
      return;
    }
    this.omitted = shoudOmit;
    
    const daikinBreadCrumbItems = this._slottedDaikinBreadCrumbItems;
    for (const [index, item] of daikinBreadCrumbItems.entries()) {
      const mode = shouldOmit ? (index === 0 || index >= daikinBreadCrumbItems.length - 2 ? "visible" : index === 1 ? "ellipsis" : "hidden") : "visible";
      item.hidden = mode === "hidden";
      item.variant = mode === "ellipsis" ? "ellipsis" : "normal";
    }
```

File: src/components/breadcrumb/daikin-breadcrumb.ts [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+  _addSlash() {
+    const daikinBreadCrumbItems = this._slottedDaikinBreadCrumbItems;
+    if (!daikinBreadCrumbItems) {
+      return;
+    }
+    daikinBreadCrumbItems.forEach((item: Element) => {
+      item.setAttribute("trailing-slash", "");
+    });
+  }
```
Comment (yodas7) [inline src/components/breadcrumb/daikin-breadcrumb.ts:125]:
```suggestion
  private _updateBreadcrumbs() {
    const daikinBreadCrumbItems = this._slottedDaikinBreadCrumbItems;
    for (const [index, item] of daikinBreadCrumbItems.entries()) {
      if (index === daikinBreadCrumbItems.length - 1) {
        // last item
        item.trailingSlash = this.trailingSlash;
        item.disabled = true;
        continue;
      }
      
      item.trailingSlash = true;
    }
  }
```

File: src/components/breadcrumb/daikin-breadcrumb.ts [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+  private divOriginalWidth: number = 0;
+
+  private omissionMode: boolean = false;
+
+  get _isEllipsis() {
+    return this.overflow === "ellipsis";
+  }
+
+  _omit() {
```
Comment (yodas7) [inline src/components/breadcrumb/daikin-breadcrumb.ts:65]:
```suggestion
  private _omit() {
```

File: src/components/breadcrumb/daikin-breadcrumb.ts [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+      ?.assignedElements({ flatten: true });
+    if (!daikinBreadCrumbItems) {
+      return null;
+    }
+    return daikinBreadCrumbItems[daikinBreadCrumbItems.length - 1];
+  }
+
+  // get div element from daikin-breadcrumb
+  @query("div")
+  private _divWrap: HTMLElement | null | undefined;
```
Comment (yodas7) [inline src/components/breadcrumb/daikin-breadcrumb.ts:42]: We should use ref.

File: src/components/breadcrumb/daikin-breadcrumb.ts [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+  // get the last daikin-breadcrumb-item from shadow root
+  get _slottedLastDaikinBreadCrumbItem() {
+    const daikinBreadCrumbItems = this.shadowRoot
+      ?.querySelector("slot")
+      ?.assignedElements({ flatten: true });
+    if (!daikinBreadCrumbItems) {
+      return null;
+    }
+    return daikinBreadCrumbItems[daikinBreadCrumbItems.length - 1];
+  }
```
Comment (yodas7) [inline src/components/breadcrumb/daikin-breadcrumb.ts:38]: we should use queryAssignedElements

File: src/components/breadcrumb/daikin-breadcrumb.visual.test.ts [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    });
+
+    test("smallScreen", async ({ page }) => {
+      await page.setViewportSize({
+        width: 640,
+        height: 480,
+      });
+
+      await page.goto(baseURL);
+      console.log(baseURL);
```
Comment (yodas7) [inline src/components/breadcrumb/daikin-breadcrumb.visual.test.ts:41]:
```suggestion
```

File: src/components/breadcrumb/stories/framework-react.tsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  tagName: "daikin-breadcrumb-item",
+  elementClass: DaikinBreadcrumbItem,
+  events: {},
+});
+
+export const metadata: Meta<DaikinBreadcrumbStoryArgs> = {
+  component: ({ ...props }: DaikinBreadcrumbStoryArgs) => (
+    <ReactDaikinBreadcrumb {...props}>
+      <ReactDaikinBreadcrumbItem href="#">
+        DaikinBreadcrumbItem1
```
Comment (yodas7) [inline src/components/breadcrumb/stories/framework-react.tsx:26]:
```suggestion
        Breadcrumb Item 1
```

**Review by yodas7 — CHANGES_REQUESTED** [review]
I'm a little worried about whether the current mechanism for omitting content is sufficient (such as recalculating content size when slot content changes, or how to display the menu for omitted items).
However, I don't have a good solution, so I think it's fine to leave it as it is until the menu is implemented.

File: cspell-dictionary.txt [hunk 11]
```diff
@@ -8,6 +8,7 @@ roboto
 tailwindcss
 taze
 tseslint
+framename
```
[the remark this answers is not in the document]
Comment (yodas7) [reply cspell-dictionary.txt:12]: bump

[the remark this answers is not in the document]
Comment (rinjInTokyo) [reply cspell-dictionary.txt]: @yodas7 
removed it

**Review by yodas7 — CHANGES_REQUESTED** [review]
I also wanted to test the use of `link` slots.

**Comment by yodas7** [comment]
@rinjInTokyo 
Forgot to mention Changesets; please add a changeset.
e.g. `Add Breadcrumbs Component.`

File: tailwind.config.cjs [hunk 12]
```diff
@@ -54,6 +54,17 @@ module.exports = defineConfig({
...
+      addVariant("slotted", [
+        // `::slotted` is equivalent to `::slotted(*)`
+        "&::slotted",
+        // `& > *` is for fallback contents. See https://github.com/w3c/csswg-drafts/issues/5482.
+        "&>*",
+      ]);
+      addVariant("slotted-focus", [
+        "&::slotted(*:focus-visible)",
+        "&>*:focus-visible",
+      ]);
```
Comment (yodas7) [inline tailwind.config.cjs:67]: Good catch. To make it more versatile, we can use the following method:

```suggestion
      matchVariant("slotted", (value) => [`&::slotted(${value})`, `&>${value}`]);
```

with `slotted-[*:focus-visible]:text-daikinBlue-700`.

**Review by yodas7 — CHANGES_REQUESTED** [review]

**Review by yodas7 — APPROVED** [review]
LGTM

**Pull request merged**